src/Sulu/Bundle/DocumentManagerBundle/Reference/Command/DocumentReferencesCommand.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\DocumentManagerBundle\Reference\Command;
+
+use Jackalope\Query\Row;
+use Jackalope\Query\RowIterator;
+use PHPCR\SessionInterface;
+use Sulu\Bundle\DocumentManagerBundle\Reference\Provider\DocumentReferenceProviderInterface;
+use Sulu\Component\Content\Document\Behavior\StructureBehavior;
+use Sulu\Component\DocumentManager\Behavior\Mapping\TitleBehavior;
+use Sulu\Component\DocumentManager\Behavior\Mapping\UuidBehavior;
+use Sulu\Component\DocumentManager\DocumentManagerInterface;
+use Sulu\Component\Webspace\Manager\WebspaceManagerInterface;
+use Sulu\Component\Webspace\Webspace;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\String\Inflector\EnglishInflector;
+
+/**
+ * @internal
+ */
+final class DocumentReferencesCommand extends Command
```

Comment:
Think we should have just one command called `bin/console sulu:reference:recalculate-refernces` or something like that.

And have a `RereferenceProviderInterface` with a `getType(): string` static method I can have one or multiple per type. In our case there would be RereferenceProviderInterface for pages, articles, snippets.